### PR TITLE
mcslock: refine builtins to c11 atomic

### PR DESCRIPTION
**Pull request by zondatw**

### CODE REVIEW

File: mcslock/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 all:
-	gcc -o tests -std=gnu11 -Wall -O2 mcslock.c tests.c
+	gcc -o tests -std=c11 -Wall -O2 mcslock.c tests.c
```
Comment (jserv) [inline mcslock/Makefile]: Don't do this. We might use GNU extensions.

File: mcslock/tests.c [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 int main(void)
 {
-    mcslock_t lock;
+    _Atomic(mcslock_t) lock;
```
Comment (jserv) [inline mcslock/tests.c]: It is a bit inconvenient. Users may regard `mcslock_t` as the userspace implementation of MCS lock and use `mcslock_t` without `_Atomic`.

File: mcslock/Makefile (hunk 1 above)
Comment (zondatw) [reply mcslock/Makefile]: OK, I'm revert it.

File: mcslock/tests.c (hunk 2 above)
Comment (zondatw) [reply mcslock/tests.c]: Yes, that's inconvenient, and  I moved definition of _Atomic for lock from test.c to msclock.h. Currently, users can use `mcslock_t` without `_Atomic`.

**Comment by jserv** [comment]
Thank @zondatw for contributing!

**Pull request merged**